test/spec/PanelHeaderProvider.spec.js

```diff
@@ -0,0 +1,271 @@
+
+      // then
+      expect(typeNode).to.exist;
+      expect(typeNode.innerText).to.equal('DECISION');
+    });
+
+  });
+
+
+  describe('#getElementLabel', function() {
+
+    it('should get label - name', function() {
+
+      // given
+      const element = createElement('dmn:KnowledgeSource', {
+        name: 'foo'
+      });
+
+      // when
+      const elementLabel = PanelHeaderProvider.getElementLabel(element);
+
+      // then
+      expect(elementLabel).to.equal('foo');
+    });
+
+
+    it('should get label - text', function() {
+
+      // given
+      const element = createElement('dmn:TextAnnotation', {
+        text: 'foo'
+      });
+
+      // when
+      const elementLabel = PanelHeaderProvider.getElementLabel(element);
+
+      // then
+      expect(elementLabel).to.equal('foo');
+    });
+
+  });
+
+
+  describe('#getConcreteType', function() {
+
+    it('should get type - decision', async function() {
+
+      // given
+      const element = createElement('dmn:Decision');
+
+      // when
+      const type = getConcreteType(element);
+
+      // then
+      expect(type).to.equal('Decision');
+    });
+
+
+    it('should get type - knowledge source', async function() {
+
+      // given
+      const element = createElement('dmn:KnowledgeSource');
+
+      // when
+      const type = getConcreteType(element);
+
+      // then
+      expect(type).to.equal('KnowledgeSource');
+    });
+
+
+    it('should get type - business knowledge model', async function() {
+
+      // given
+      const element = createElement('dmn:BusinessKnowledgeModel');
+
+      // when
+      const type = getConcreteType(element);
+
+      // then
+      expect(type).to.equal('BusinessKnowledgeModel');
+    });
+
+
+    it('should get type - input data', async function() {
+
+      // given
+      const element = createElement('dmn:InputData');
+
+      // when
+      const type = getConcreteType(element);
+
+      // then
+      expect(type).to.equal('InputData');
+    });
+
+
+    it('should get type - information requirement', async function() {
+
+      // given
+      const sequenceFlow = createElement('dmn:InformationRequirement');
+
+      // when
+      const type = getConcreteType(sequenceFlow);
+
+      // then
+      expect(type).to.equal('InformationRequirement');
+    });
+
+
+    it('should get type - authority requirement', async function() {
+
+      // given
+      const sequenceFlow = createElement('dmn:AuthorityRequirement');
+
+      // when
+      const type = getConcreteType(sequenceFlow);
+
+      // then
+      expect(type).to.equal('AuthorityRequirement');
+    });
+
+
+    it('should get type - knowledge requirement', async function() {
+
+      // given
+      const sequenceFlow = createElement('dmn:KnowledgeRequirement');
+
+      // when
+      const type = getConcreteType(sequenceFlow);
+
+      // then
+      expect(type).to.equal('KnowledgeRequirement');
+    });
```

Comment:
More new lines please. 😇 

```javascript
    });
    
  });

});
    
```

Reply:
It's all good. Petty makes pretty

Reply:
I pushed changes for this round of feedback (newlines, indentation and revising connections)